core/app/models/mno_enterprise/tenant_config.rb

```diff
@@ -121,6 +113,39 @@ def self.build_object(schema)
       end
     end
 
+    # Flag field as readonly in the config schema
+    # A field is readonly if the `<field>_readonly` key is `true` in the `frontend_config`
+    def self.update_readonly_fields(schema, frontend_config)
```

Comment:
I'd recommend changing the method name to `mark_readonly_fields` or `add_readonly_flags` or `add_readonly_properties`

The name `update_readonly_fields` sounds weird for some reasons 😃